compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -899,6 +899,10 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             return;
         }
 
+        if obligation.references_error() {
```

Comment:
Can you move this check into `confirm_transmutability_candidate`? I think it's better to put this check closer to where the original ICE was happening.

Also, is it the `obligation.param_env` that references error, or the `obligation.predicate`? I would appreciate this `if` condition to be as specific as possible.

Also, can you comment why it's needed?

Reply:
I have changed the return type of `Assume::from_const` to `Option<Self>`.